### PR TITLE
Feat iterative match

### PR DESCRIPTION
**Pull request by davidsantiagoquevedo**
This PR introduces an iterative matching routine to avoid underestimating the VE when removing unvaccinated outcomes whose exposure times do not coincide with their vaccinated pair. Three main functions were added:

1. `iterate_match()`: Handles the main while loop to search for new pairs. It rematches the population separately, first removing vaccinated individuals and then removing unvaccinated individuals. The loop continues until there are no more matches to perform or a certain threshold is reached.
2. `rematch_()`: Controls the re-matching of removed pairs. A `tryCatch` statement is used to handle possible errors.
3. `capture_warnings()`: Captures warning messages to handle tests and to report in the summary of the matching routine.

By introducing these functions, the pipeline of `match_static()` was changed. The iterative match is performed after the initial adjustment. The row `removed` was also deprecated from `match_summary()`.

Necessary tests were built for all functions.

### CODE REVIEW

File: R/coh_match_iterate.R [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  adjusted_i <- adjusted
+  removed_i <- matched_i[!(matched_i$match_id %in% adjusted_i$match_id), ]
+
+  # Set control parameter to avoid infinte while loop
+  im <- 0
+  thershold <- nrow(removed_i) #maximum number of iterations: size of removed
+
+  # Iterate until threshold
+  # or until the procedure generates zero adjusted population
+  while ((nrow(adjusted_i) != 0) && (im < thershold)) { #nolint
```
Comment (GeraldineGomez) [inline R/coh_match_iterate.R:143]: Removing #nolint comments is necessary for CRAN submission

File: R/handler_warnings.R [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#' @title Capture Warnings
+#'
+#' @description This function captures the warnings generated by an
+#' expression without printing them to the console. It returns both the
+#' result of the expression and the captured warnings. It is useful for
+#' avoiding warnings appearing in the middle of a routine (for instance,
```
Comment (GeraldineGomez) [inline R/handler_warnings.R:6]: This is useful for 
#' preventing warnings from appearing during routines, such as
#' iterative processes

File: R/handler_warnings.R [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#' @keywords internal
+
+capture_warnings <- function(expr, warn_env = new.env()) {
+  warn_env$warnings <- character()
+
+  result <- withCallingHandlers(
+    expr,
+    warning = function(w) {
+      warn_env$warnings <- c(warn_env$warnings, conditionMessage(w))
+      invokeRestart("muffleWarning")
```
Comment (GeraldineGomez) [inline R/handler_warnings.R:25]: Santi, please validate the use of  ` invokeRestart("muffleWarning")`  according to CRAN policy

File: R/handler_warnings.R [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#' @title Capture Warnings
+#'
+#' @description This function captures the warnings generated by an
+#' expression without printing them to the console. It returns both the
+#' result of the expression and the captured warnings. It is useful for
+#' avoiding warnings appearing in the middle of a routine (for instance,
+#' an iterative routine).
+#'
+#' @param expr An expression to evaluate.
+#' @param warn_env An optional environment to store the warnings.
```
Comment (GeraldineGomez) [inline R/handler_warnings.R:10]: Specify that the function does not affect the user's environment and explain the usage of `invokeRestart("muffleWarning")`

**Review by GeraldineGomez — CHANGES_REQUESTED** [review]

File: R/coh_match.R [hunk 5]
```diff
@@ -208,11 +210,12 @@ match_cohort <- function(data_set,
 #' @description Summarizes the results of `match_cohort`.
 #'
 #' @param object Object of the class `match`.
+#' @param warnings_log If `TRUE`, prints the warnings log.
 #' @param ... Additional arguments passed to other functions.
```
[the remark this answers is not in the document]
Comment (davidsantiagoquevedo) [reply R/coh_match.R:214]: I included an example using `warnings_log` in the documentation of `match_cohort()`

**Review by GeraldineGomez — APPROVED** [review]

**Pull request merged**